File: go-service/models/tax_return.go
package models

import "time"

// TaxReturn represents a tax return submission
type TaxReturn struct {
	ID                int        `db:"Id" json:"id"`
	Period            string     `db:"Period" json:"period"`
	TaxType           string     `db:"TaxType" json:"taxType"`
	Amount            float64    `db:"Amount" json:"amount"`
	Status            string     `db:"Status" json:"status"`
	DueDate           time.Time  `db:"DueDate" json:"dueDate"`
	SubmittedDate     *time.Time `db:"SubmittedDate" json:"submittedDate,omitempty"`
	SubmittedByUserID *int       `db:"SubmittedByUserId" json:"submittedByUserId,omitempty"`
	ReferenceNumber   *string    `db:"ReferenceNumber" json:"referenceNumber,omitempty"`
	CreatedAt         time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt         *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

File: frontend/src/app/core/domain/tax-return/tax-return.ts
import { User } from '../domain.barrel';
import { TaxReturnStatus, TaxType } from '../../enums/enums.barrel';

/**
 * Represents a Tax Return
 */
export interface TaxReturn {
  /** ID */
  id: number;

  /** Period - e.g., "December 2023" */
  period: string;

  /** Tax Type */
  taxType: TaxType;

  /** Amount */
  amount: number;

  /** Status */
  status: TaxReturnStatus;

  /** Due Date */
  dueDate: Date;

  /** Submitted Date */
  submittedDate?: Date | null;

  /** Submitted By User ID - FK to User */
  submittedByUserId?: number | null;

  /** Reference Number */
  referenceNumber?: string | null;

  /** Submitted By User */
  submittedByUser?: User | null;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
